### ISSUE TITLE
Keeps on trying to sign-in 

### ISSUE THREAD

**Issue by bhaidar** [damienbod/angular-auth-oidc-client#254]
Hi @damienbod 
I am integrating the OIDC in my Angular 5 app and for some reason, it keeps on trying to sign in the user non-stop redirects.

The application is structured with 2 major modules in mind:
- **RootModule**: The Main module that contains `APP_INITIALIZER` code to call STS server and load other data related to the application.
- **AppModule**: It has everything related to the business application

I've grouped all OIDC stuff in a single module **AuthModule**.

````TypeScript
import {
  AuthModule as OidcAuthModule,
  OidcSecurityService,
  OpenIDImplicitFlowConfiguration,
  OidcConfigService,
  AuthWellKnownEndpoints
} from 'angular-auth-oidc-client';

import { AppConsts } from '@common/models';

export const AUTH_COMPONENTS = [
  AuthCallbackComponent,
  UnauthorizedComponent
];

export const AUTH_SERVICES = [
  AuthorizationCanGuard,
  AuthorizationGuard,
  OidcConfigService,
  OidcSecurityService
];

@NgModule({
  imports: [
    CommonModule,
    BrowserAnimationsModule,
    FormsModule,
    ReactiveFormsModule,
    HttpModule,
    HttpClientModule,
    OidcAuthModule.forRoot()
  ],
  declarations: [
    ...AUTH_COMPONENTS
  ],
  exports: [
    ...AUTH_COMPONENTS
  ]
})
export class AuthModule {

  static forRoot(): ModuleWithProviders {
    return {
      ngModule: AuthModule,
      providers: [
        ...AUTH_SERVICES
      ]
    };
  }

  constructor(
    private oidcSecurityService: OidcSecurityService,
    private oidcConfigService: OidcConfigService
  ) {
    this.oidcConfigService.onConfigurationLoaded.subscribe(() => {

      const openIDImplicitFlowConfiguration = new OpenIDImplicitFlowConfiguration();
      openIDImplicitFlowConfiguration.stsServer = AppConsts.stsServer.url;

      // The Client MUST validate that the aud (audience) Claim contains its client_id value 
      // registered at the Issuer identified by the iss (issuer) Claim as an audience.
      // The ID Token MUST be rejected if the ID Token does not list the Client as a valid audience, 
      // or if it contains additional audiences not trusted by the Client.

      openIDImplicitFlowConfiguration.client_id = AppConsts.stsServer.clientId;
      openIDImplicitFlowConfiguration.redirect_url = AppConsts.stsServer.redirectUrl;
      openIDImplicitFlowConfiguration.post_logout_redirect_uri = AppConsts.stsServer.postLogoutRedirectUri;

      openIDImplicitFlowConfiguration.response_type = 'id_token token';
      openIDImplicitFlowConfiguration.scope = 'openid profile';
      openIDImplicitFlowConfiguration.trigger_authorization_result_event = true;

      openIDImplicitFlowConfiguration.silent_renew = false;

      openIDImplicitFlowConfiguration.post_login_route = '/callback';
      // HTTP 403
      openIDImplicitFlowConfiguration.forbidden_route = '/forbidden';
      // HTTP 401
      openIDImplicitFlowConfiguration.unauthorized_route = '/unauthorized';
      openIDImplicitFlowConfiguration.log_console_warning_active = true;
      openIDImplicitFlowConfiguration.log_console_debug_active = false;
      // id_token C8: The iat Claim can be used to reject tokens that were issued too far away from the current time,
      // limiting the amount of time that nonces need to be stored to prevent attacks.The acceptable range is Client specific.
      openIDImplicitFlowConfiguration.max_id_token_iat_offset_allowed_in_seconds = 60;

      openIDImplicitFlowConfiguration.storage = localStorage;

      const authWellKnownEndpoints = new AuthWellKnownEndpoints();
      authWellKnownEndpoints.setWellKnownEndpoints(this.oidcConfigService.wellKnownEndpoints);

      this.oidcSecurityService.setupModule(openIDImplicitFlowConfiguration, authWellKnownEndpoints);
    });
  }
}
````
In the **RootModule** I import the following:

````TypeScript
AuthModule.forRoot(), // handles oidc authentication
AuthModule, // for components
````
In the **AppModule** I don't import anything however, I make use of the Guards to protect the routes in the **AppRoutingModule**.

In summary, the **RootModule** loads both **AuthModule** and **AppModule** eagerly.

Now, when I run the application, I am redirected to the IDP. I enter my credentials and I am redirected back to my app. I see the Authorization Results of 1, but then, it says, Logging in and the loop starts and never ends!

I can't any errors logged to share here.

Any idea if the setup is missing something?

Many thanks
Bilal

**Comment by bhaidar** [comment]
I might have solved the problem.

Before refactoring to using **AuthModule**, I was subscribing to the `oidcConfigService.onConfigurationLoaded` in the RootModule. So it seems I kept the configuration there and this is what was causing the loop.

I will do more tests to confirm.

Thanks
Bilal

**Comment by jhossy** [comment]
@bhaidar Did you solve it? I also have a redirect loop so it might be a similar issue as the one you experience

**Comment by bhaidar** [comment]
@jhossy 
Yes, I had the module configured twice in the application. That's all, I remove one of the implementations and kept the other.

What's going on with you? Maybe I can assist.

Regards
Bilal

**Comment by jhossy** [comment]
@bhaidar Actually, it might have been something similar in my case. I got around it shortly after posting here, but thanks for answering :)
